### PR TITLE
Support Meteor 1.8.1

### PR DESCRIPTION
**Pull request by zodern**
Tests pass locally on both Meteor 1.8 and 1.8.1.

### CODE REVIEW

File: server.coffee [hunk 1]
```diff
@@ -136,14 +148,21 @@ extendPublish (name, publishFunction, options) ->
...
         # with new values. Only top-level fields matter.
-        for field of @_session.getCollectionView(collectionName)?.documents?[stringId]?.dataByKey or {}
+        _documents = @_session.getCollectionView(collectionName)?.documents or {}
+        if _documents instanceof Map
+          dataByKey = _documents.get(stringId)?.dataByKey or {}
+        else
+          dataByKey = _documents?[stringId]?.dataByKey or {}
+
+        iterateObjectOrMapKeys dataByKey, (field) =>
           oldFields[field] = undefined
```
Comment (mitar) [inline server.coffee:164]: Are you sure in Meteor 1.8 they are still setting it to undefined?

File: package.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ Package.onTest(function (api) {
     'peerlibrary:assert@0.2.5',
     'peerlibrary:server-autorun@0.7.1',
     'peerlibrary:classy-test@0.3.0',
-    'meteorhacks:unblock@1.1.0'
+    'lamhieu:unblock@1.0.0'
```
Comment (mitar) [inline package.js:57]: What is this change?

Comment (zodern) [reply package.js:57]: `meteorhacks:meteorx` which is used by `meteorhacks:unblock` doesn't work with Meteor 1.8.1. I've been using `lamhieu:meteorx` and `lamhieu:unblock` in Monti APM with no problems so far.

File: server.coffee (hunk 1 above)
Comment (zodern) [reply server.coffee:164]: Yes, that did not change in Meteor 1.8.1.

**Comment by mitar** [comment]
Great. Thanks. I would like to get one more confirmation from somebody that this work (I am unable to run this at the moment) and then I can merge and release it.

**Comment by brucejo75** [comment]
Hi all,

I took the PR for a spin and it fixes my case.  I can see the `"removed"` message coming through now.

Thanks much!

**Comment by etyp** [comment]
@mitar we have tested it in our beta environment (basically a replication of production) and it's working well so far.

**Comment by etyp** [comment]
@mitar given that it's working for us and @brucejo75 was also able to confirm, is there anything else you'd like to test or ensure before you feel comfortable merging and releasing a new version?

**Comment by mitar** [comment]
I will try to get it out today.

**Comment by wildhart** [comment]
Works for me!  Tested with Meteor 1.8.1-beta.17 and 1.8.1.

Thanks @zodern!

**Comment by mitar** [comment]
Published as `peerlibrary:reactive-publish@0.8.0`.

**Pull request merged**